pkg/kubelet/kubelet_node_status.go

```diff
@@ -238,13 +238,24 @@ func (kl *Kubelet) initialNode() (*v1.Node, error) {
 		Effect: v1.TaintEffectNoSchedule,
 	}
 
-	// If TaintNodesByCondition enabled, taint node with TaintNodeUnschedulable when initializing
+	notReadyTaint := v1.Taint{
+		Key:    schedulerapi.TaintNodeNotReady,
+		Effect: v1.TaintEffectNoSchedule,
+	}
+
+	// If TaintNodesByCondition enabled, taint node with TaintNodeUnschedulable and TaintNodeNotReady when initializing
 	// node to avoid race condition; refer to #63897 for more detail.
 	if utilfeature.DefaultFeatureGate.Enabled(features.TaintNodesByCondition) {
 		if node.Spec.Unschedulable &&
 			!taintutil.TaintExists(nodeTaints, &unschedulableTaint) {
 			nodeTaints = append(nodeTaints, unschedulableTaint)
 		}
+		// Add not-ready taint while initializing the node to avoid waiting for node controller to apply the taint.
+		// This taint can be removed by node lifecycle controller once node condition Type is `Ready` and Status is
+		// `True`
+		if !taintutil.TaintExists(nodeTaints, &notReadyTaint) {
```

Comment:
@ravisantoshgudimetla I also had this idea prior to bringing up #72315 :) The only concern on this approach is that the change is applied on kubelet, which means end-user has to upgrade their data plane - which is a big effort for cluster administrating, and also we need to cosnider version skewing.